src/cache-worker.ts

```diff
@@ -21,20 +21,39 @@
  */
 /* eslint-disable import/no-extraneous-dependencies*/
 import { GraphQLResponse } from 'apollo-server-types'
-import { GraphQLError } from 'graphql'
 import { GraphQLClient, gql } from 'graphql-request'
 import jwt from 'jsonwebtoken'
+import { splitEvery } from 'ramda'
 
-import { wait } from './utils'
+import { AbstractCacheWorker, Task } from './types'
+import { wait, Stack } from './utils'
 
-export class CacheWorker {
+/**
+ * Cache Worker of Serlo's API
+ * makes the API to cache values of some chosen keys.
+ * The user has to edit the file cache-keys.json for that.
+ * Add environment variable PAGINATION in order to detemine
+ * how many keys are going to be requested to be updated
+ * each time
+ */
+export class CacheWorker implements AbstractCacheWorker {
   private grahQLClient: GraphQLClient
 
+  /** The successful responses from Serlo's API */
   public okLog: GraphQLResponse[] = []
+  /**
+   *  The errors that ocurred while trying
+   *  to update values of given keys.
+   *  They can be caused by the client or
+   *  by the server as well as from other
+   *  origins.
+   */
   public errorLog: Error[] = []
 
   private pagination: number
 
+  private tasks!: Stack<Task>
```

Comment:
```suggestion
  private tasks = []
```
Then you can use the same implementation in bisect() and makeStackOutOfKeys()